server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -135,7 +136,28 @@ public boolean supportsSampling() {
     }
 
     @Override
-    public boolean supportsParallelCollection() {
+    public boolean supportsParallelCollection(ToLongFunction<String> fieldCardinalityResolver) {
+        /*
+         * we parallelize only if the cardinality of the field is lower than shard size, this is to minimize precision issues.
+         * When ordered by term, we still take cardinality into account to avoid overhead that concurrency may cause against
+         * high cardinality fields.
+         */
+        if (script() == null
+            && (executionHint == null || executionHint.equals(TermsAggregatorFactory.ExecutionMode.GLOBAL_ORDINALS.toString()))) {
+            long cardinality = fieldCardinalityResolver.applyAsLong(field());
+            if (cardinality != -1) {
+                if (InternalOrder.isKeyOrder(order)) {
+                    if (cardinality <= 50) {
```

Comment:
Maybe turn `50` into a constant?

Reply:
I will do this as soon as I make the same change for composite aggs, next round.